crates/bevy_asset/src/path.rs

```diff
@@ -290,6 +290,124 @@ impl<'a> AssetPath<'a> {
         self.clone().into_owned()
     }
 
+    /// Resolves a relative asset path via concatenation. The result will be an `AssetPath` which
+    /// is resolved relative to this "base" path. There are several cases:
+    ///
+    /// If the `path` argument begins with `#`, then it is considered an asset label, in which case
+    /// the result is the base path with the label portion replaced.
+    ///
+    /// If the path argument begins with '/', then it is considered an 'full' path, in which
+    /// case the result the result is a new `AssetPath` consisting of the base path asset source
+    /// (if any) with the path and label portions of the relative path. Note that a 'full'
+    /// asset path is still relative to the asset source root, and not necessarily an absolute
+    /// filesystem path.
+    ///
+    /// If the `path` argument begins with an asset source (ex: `http://`) then the entire base
+    /// path is replaced - the result is the source, path and label (if any) of the `path`
+    /// argument.
+    ///
+    /// Otherwise, the `path` argument is considered a relative path. The result is concatenated
+    /// using the following algorithm:
+    ///
+    /// * The base path and the `path` argument are concatenated.
+    /// * Path elements consisting of "/." or "&lt;name&gt;/.." are removed.
+    ///
+    /// If there are insufficient segments in the base path to match the ".." segments,
+    /// then any left-over ".." segments are left as-is.
+    pub fn resolve<'b>(&'a self, path: &'b str) -> Result<AssetPath<'a>, ParseAssetPathError> {
+        self.resolve_internal(path, false)
+    }
+
+    /// Resolves an embedded asset path via concatenation. The result will be an `AssetPath` which
+    /// is resolved relative to this path. This is similar in operation to `resolve`, except that
+    /// the the 'file' portion of the base path (that is, any characters after the last '/')
+    /// is removed before concatenation, in accordance with the behavior specified in
+    /// IETF RFC 1808 "Relative URIs".
+    ///
+    /// The reason for this behavior is that embedded URIs which start with "./" or "../" are
+    /// relative to the *directory* containing the asset, not the asset file. This is consistent
+    /// with the behavior of URIs in `JavaScript`, CSS, HTML and other web file formats. The
+    /// primary use case for this method is resolving relative paths embedded within asset files,
+    /// which are relative to the asset in which they are contained.
+    ///
+    /// So for example, the path `"x/y/z#foo"` combined with a relative path of `"./a#bar"`
+    /// yields `"x/y/a#bar"`.
+    ///
+    /// Paths beginning with '/' or '#' are handled exactly the same as with `resolve()`.
+    pub fn resolve_embed<'b>(
+        &'a self,
+        path: &'b str,
+    ) -> Result<AssetPath<'a>, ParseAssetPathError> {
+        self.resolve_internal(path, true)
+    }
+
+    fn resolve_internal<'b>(
```

Comment:
Given that this is creating owned values, I think we should just return `AssetPath<'static>` for maximum flexibility.